src/main/java/org/jenkinsci/plugins/workflow/multibranch/SCMBinder.java

```diff
@@ -51,12 +51,17 @@
 import org.jenkinsci.plugins.workflow.job.WorkflowRun;
 
 /**
- * Checks out the desired version of {@link WorkflowBranchProjectFactory#SCRIPT}.
+ * Checks out the desired version of the script referred to by scriptPath.
  */
 class SCMBinder extends FlowDefinition {
 
     /** Kill switch for JENKINS-33273 in case of problems. */
     static /* not final */ boolean USE_HEAVYWEIGHT_CHECKOUT = Boolean.getBoolean(SCMBinder.class.getName() + ".USE_HEAVYWEIGHT_CHECKOUT"); // TODO 2.4+ use SystemProperties
+    private String scriptPath;
```

Comment:
You need to either set this to `SCRIPT` in a `readResolve`, or make sure that all calls are prepared for it to be null. Otherwise you will be breaking existing branch projects after a plugin update.